source/plugins/go/src/extension/extension.go

```diff
@@ -40,64 +39,73 @@ func (e *Extension) GetOutputStreamId(datatype string) string {
+	if err != nil {
+		message := fmt.Sprintf("Error getting datatype to named pipe mapping: %s", err.Error())
+		logger.Printf(message)
+	}
+	return e.datatypeNamedPipeMap[datatype]
+}
+
+func getExtensionDataTypeMapping(isNamedPipe bool) (map[string]string, error) {
 	guid := uuid.New()
-	datatypeOutputStreamMap := make(map[string]string)
+	datatypeMap := make(map[string]string)
 
 	taggedData := map[string]interface{}{"Request": "AgentTaggedData", "RequestId": guid.String(), "Tag": "ContainerInsights", "Version": "1"}
 	jsonBytes, err := json.Marshal(taggedData)
 	// TODO: this error is unhandled
 
-	var data []byte
-	enc := codec.NewEncoderBytes(&data, new(codec.MsgpackHandle))
-	if err := enc.Encode(string(jsonBytes)); err != nil {
-		return datatypeOutputStreamMap, err
-	}
-
-	fs := &FluentSocket{}
-	fs.sockAddress = "/var/run/mdsd-ci/default_fluent.socket"
-	if containerType != "" && strings.Compare(strings.ToLower(containerType), "prometheussidecar") == 0 {
-		fs.sockAddress = fmt.Sprintf("/var/run/mdsd-%s/default_fluent.socket", containerType)
-	}
-	responseBytes, err := FluentSocketWriter.writeAndRead(fs, data)
-	defer FluentSocketWriter.disconnect(fs)
-	logger.Printf("Info::mdsd::Making call to FluentSocket: %s to write and read the config data", fs.sockAddress)
+	response, err := getExtensionConfigResponse(jsonBytes)
 	if err != nil {
-		return datatypeOutputStreamMap, err
+		return datatypeMap, err
 	}
-	response := string(responseBytes) // TODO: why is this converted to a string then back into a []byte?
 
-	var responseObjet AgentTaggedDataResponse
-	err = json.Unmarshal([]byte(response), &responseObjet)
+	var responseObject AgentTaggedDataResponse
+	err = json.Unmarshal([]byte(response), &responseObject)
 	if err != nil {
-		logger.Printf("Error::mdsd::Failed to unmarshal config data. Error message: %s", string(err.Error()))
-		return datatypeOutputStreamMap, err
+		logger.Printf("Error::mdsd/Windows AMA::Failed to unmarshal config data. Error message: %s", string(err.Error()))
+		return datatypeMap, err
 	}
 
 	var extensionData TaggedData
-	json.Unmarshal([]byte(responseObjet.TaggedData), &extensionData)
+	json.Unmarshal([]byte(responseObject.TaggedData), &extensionData)
 
 	extensionConfigs := extensionData.ExtensionConfigs
-	logger.Printf("Info::mdsd::build the datatype and streamid map -- start")
+	outputStreamDefinitions := make(map[string]StreamDefinition)
+	if isNamedPipe == true {
+		outputStreamDefinitions = extensionData.OutputStreamDefinitions
+	}
+	logger.Printf("Info::mdsd/Windows AMA::build the datatype and streamid map -- start")
 	for _, extensionConfig := range extensionConfigs {
 		outputStreams := extensionConfig.OutputStreams
 		for dataType, outputStreamID := range outputStreams {
 			logger.Printf("Info::mdsd::datatype: %s, outputstreamId: %s", dataType, outputStreamID)
-			datatypeOutputStreamMap[dataType] = outputStreamID.(string)
+			if isNamedPipe {

```

Comment:
instead of "isNamedPipe", can we say "isWindows" or "isLinux" and make decision based on the platform?

Reply:
The flag is not for WIndows v/s Linux. It is to get named pipe v/s stream id.